base/compiler/ssair/ir.jl

```diff
@@ -1564,7 +1557,7 @@ end
 
 function complete(compact::IncrementalCompact)
     result_bbs = resize!(compact.result_bbs, compact.active_result_bb-1)
-    cfg = CFG(result_bbs, Int[first(result_bbs[i].stmts) for i in 2:length(result_bbs)])
```

Comment:
It seems like we were adding extra complexity here and also at the `CFG` construction within `domsort_ssa!`? Now I'm seeing very few cases causing `BoundsError` that seem to be caused by these changes.

My question is, can't `block_for_inst(index::Vector{Int}, inst::Int)` be used in place of `block_for_inst(index::Vector{BasicBlock}, inst::Int)` interchangeably? If not, I think we may still want to keep maintaining `CFG.index` field.